src/SnowpackCompactionAR2023Mod.F90

```diff
@@ -0,0 +1,147 @@
+module SnowpackCompactionARMod
```

Comment:
It's better to have a consistent file name and module name, which are consistent with other codes.
e.g., change SnowpackCompactionAR2023Mod.F90 to SnowpackCompactionARMod.F90